dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,336 @@
+                .andExpect(jsonPath("$[3].type",
+                        Matchers.hasToString(orcidValue)));
+
+        DSpaceServicesFactory.getInstance().getConfigurationService().reloadConfig();
+        metadataAuthorityService.clearCache();
+        choiceAuthorityService.clearCache();
+    }
+
+    @Test
+    public void findTypedLinkForBitstream() throws Exception {
+        String bitstreamContent = "ThisIsSomeDummyText";
+        String bitstreamMimeType = "text/plain";
+
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Item Test")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .build();
+
+        Bitstream bitstream = null;
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+            bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream")
+                    .withDescription("description")
+                    .withMimeType(bitstreamMimeType)
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        String uiUrl = configurationService.getProperty("dspace.ui.url");
+        String serverUrl = configurationService.getProperty("dspace.server.url");
+        getClient().perform(get("/signposting/links/" + bitstream.getID())
+                        .header("Accept", "application/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",
+                        Matchers.hasSize(3)))
+                .andExpect(jsonPath("$[0].href",
+                        Matchers.hasToString(uiUrl + "/entities/publication/" + item.getID())))
+                .andExpect(jsonPath("$[0].rel",
+                        Matchers.hasToString("collection")))
+                .andExpect(jsonPath("$[0].type",
+                        Matchers.hasToString("text/html")))
+                .andExpect(jsonPath("$[1].href",
+                        Matchers.hasToString(serverUrl + "/signposting/linksets/" + item.getID())))
+                .andExpect(jsonPath("$[1].rel",
+                        Matchers.hasToString("linkset")))
+                .andExpect(jsonPath("$[1].type",
+                        Matchers.hasToString("application/linkset")))
+                .andExpect(jsonPath("$[2].href",
+                        Matchers.hasToString(serverUrl + "/signposting/linksets/" + item.getID() + "/json")))
+                .andExpect(jsonPath("$[2].rel",
+                        Matchers.hasToString("linkset")))
+                .andExpect(jsonPath("$[2].type",
+                        Matchers.hasToString("application/linkset+json")));
+
+        DSpaceServicesFactory.getInstance().getConfigurationService().reloadConfig();
+        metadataAuthorityService.clearCache();
+        choiceAuthorityService.clearCache();
+    }
+
```

Comment:
We appear to be missing tests for `BitstreamLicenseProcessor` and `ItemLicenseProcessor`.  I'm also not able to find a way to get licenses to appear from `/linksets/[uuid]/json`... so I'm not sure they are working.

We also need tests of `/signposting/linksets/[item-uuid]/json` for the following item types:
* An Item that is in Workspace or Workflow
* An Item that is withdrawn
* An Item that is currently under embargo
* An Item that is access restricted (e.g. to Administrators group)

Similarly we should add tests to `/signposting/links/[bitstream-uuid]` for similar types of Bitstreams:
* A Bitstream which is restricted
* A Bitstream which is embargoed
* A Bitstream of a WorkspaceItem, WorkflowItem or Withdrawn or Embargoed Item.

Essentially we don't appear to have any tests to verify access permissions are working appropriately.

Reply:
Additional tests has been added.